src/mlpack/core/data/load_image_impl.hpp

```diff
@@ -0,0 +1,219 @@
+/**
+ * @file load_image_impl.hpp
+ * @author Mehul Kumar Nirala
+ *
+ * An image loading utillity
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_CORE_DATA_LOAD_IMAGE_IMPL_HPP
+#define MLPACK_CORE_DATA_LOAD_IMAGE_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "load_image.hpp"
+
+namespace mlpack {
+namespace data {
+
+LoadImage::LoadImage():
+        matrixWidth(0),
+        matrixHeight(0),
+        channels(3)
+{
+  fileTypes.clear();
+
+  // Declare supported file types.
+  fileTypes.insert(fileTypes.end(),
+    {"jpg", "png", "tga", "bmp", "psd", "gif", "hdr", "pic", "pnm"});
+}
+
+LoadImage::LoadImage(size_t width,
+          size_t height,
+          size_t channels):
+          matrixWidth(width),
+          matrixHeight(height),
+          channels(channels)
+{
+  fileTypes.clear();
+
+  // Declare supported file types.
+  fileTypes.insert(fileTypes.end(),
+    {"jpg", "png", "tga", "bmp", "psd", "gif", "hdr", "pic", "pnm"});
+}
+
+LoadImage::~LoadImage()
+{
+  // Do nothing.
+}
+
+bool LoadImage::ImageFormatSupported(const std::string& fileName)
+{
+  // Iterate over all supported file types.
+  for (auto extension : fileTypes)
+    if (extension == Extension(fileName))
+      return true;
+  return false;
+}
+
+bool LoadImage::Load(const std::string& fileName,
+                      bool flipVertical,
+                      arma::Mat<unsigned char>&& outputMatrix,
```

Comment:
Did you see my comment about using a reference and the idea behind ``width, height, channels`` if we load images of different sizes.

Reply:
@zoq Yes, however, there are some issues associated with resizing of an image via stb.

Reply:
Not sure, do we want the option to resize images?

Reply:
> > Added functionality for loading images with the help of [stb_image.h](https://github.com/nothings/stb/blob/master/stb_image.h).
> > Currently supports `jpg`,`png`,`tga`, `bmp`, `psd`, `gif`, `hdr`, `pic`, `pnm`.
> 
> @MuLx10 Just curious , suppose just like loading images ,can we save too using this.? For suppose we have mat and simply we can save it as an image.?

Now it can save image as well :smile:

Reply:
@zoq not sure what you meant.